src/modules/RA/DataLayer/components/DataLayerTabbedForm.js

```diff
@@ -53,276 +52,300 @@ const LazyFormTab = ({ hidden, ...props }) => (
+                            validate={defaultRequired}
                           />
-                        )}
-                      </>
-                    );
-                  }}
-                </FormDataConsumer>
-                {formData.table_enable ? <BooleanInput source="shown" label="datalayer.form.show" /> : <></>}
-                <FormDataConsumer>
-                  {({ scopedFormData = {}, getSource }) => (
-                    scopedFormData.shown && <BooleanInput source={getSource('display')} label="Afficher ce champs par défaut" />
-                  )}
-                </FormDataConsumer>
-                {formData.table_export_enable ? <BooleanInput source="exportable" label="datalayer.form.exportable" /> : <></>}
-              </DraggableFormIterator>
-            </ArrayInput>
-          )}
-        </FormDataConsumer>
-      </LazyFormTab>
-    </TabbedForm>
-  </>
-);
+                          {filterType && (
+                            <BooleanInput
+                              source={getSource('filter_settings.fetchValues')}
+                              label="datalayer.form.type.fetch.label"
+                              fullWidth
+                            />
+                          )}
+                          {(filterType && !filterFetch) && (
+                            <TextArrayInput
+                              source={getSource('filter_settings.values')}
+                              label={`datalayer.form.type.values${filterType === 'many' ? '' : '_optional'}`}
+                              validate={filterType === 'many' ? defaultRequired : undefined}
+                              fullWidth
+                            />
+                          )}
+                          {filterType === 'range' && (
+                            <SelectInput
+                              source={getSource('filter_settings.format')}
+                              label="datalayer.form.type.range_format.label"
+                              choices={[
+                                { id: 'number', name: 'datalayer.form.type.range_format.number' },
+                                { id: 'date', name: 'datalayer.form.type.range_format.date' },
+                              ]}
+                            />
+                          )}
+                        </>
+                      );
+                    }}
+                  </FormDataConsumer>
+                  {formData.table_enable ? <BooleanInput source="shown" label="datalayer.form.show" /> : <></>}
+                  <FormDataConsumer>
+                    {({ scopedFormData = {}, getSource }) => (
+                      scopedFormData.shown && <BooleanInput source={getSource('display')} label="Afficher ce champs par défaut" />
+                    )}
+                  </FormDataConsumer>
+                  {formData.table_export_enable ? <BooleanInput source="exportable" label="datalayer.form.exportable" /> : <></>}
```

Comment:
```suggestion
                  {formData.table_export_enable && <BooleanInput source="exportable" label="datalayer.form.exportable" />}
```

Reply:
Not really in scope of the `react-admin` v2 ➡ v3 migration.

Anyway, `<DraggableFormIterator />` does not accept `null` as child *(it requires strictly a component as child, and `null`is not)*. It may be patched to avoid error when it happens.

Reply:
Also in some cases, null is not a valid for child for some specific components. Returning an empty Fragment is then a quite simple security measure.